services/head_tracker.go

```diff
@@ -193,7 +193,7 @@ func (ht *HeadTracker) subscribeToNewHeads(headers chan models.BlockHeader) (mod
 func (ht *HeadTracker) updateBlockHeader() {
 	header, err := ht.store.TxManager.GetBlockByNumber("latest")
 	if err != nil {
-		logger.Warnw("Unable to update latest block header", "err", err)
+		logger.Errorw("Unable to update latest block header", "err", err)
```

Comment:
Optional nitpick: The functional difference between `Warn` and `Error` is that `Error` will dump out the stack trace. That can get spammy very quickly, but I understand the desire to have all the information if we ever need to debug, and an error label might be nicer here. There are a few here I would leave as `Warn` , like this one.